File: 2586-count-the-number-of-vowel-strings-in-range/2586-count-the-number-of-vowel-strings-in-range.js
/**
 * @param {string[]} words
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var vowelStrings = function(words, left, right) {
    const vowels = new Set([..."aeiou"])
    let res = 0
    for (let i=left; i <= right; i++) {
        if (vowels.has(words[i].at(0)) && vowels.has(words[i].at(-1))) {
            res++
        }
    }
    return res
};

File: 2586-count-the-number-of-vowel-strings-in-range/2586-count-the-number-of-vowel-strings-in-range.py
class Solution:
    def __init__(self):
        self.vowels = "aeiou"
    
    def is_valid(self, word: str) -> bool:
        return word[0] in self.vowels and word[-1] in self.vowels
    
    def vowelStrings(self, words: List[str], left: int, right: int) -> int:
        return sum([1 for word in words[left:right+1:] if word[0] in 'aeiou' and word[-1] in 'aeiou'])
    
        #
        
        res = 0
        for i in range(left, right+1):
            if self.is_valid(words[i]):
                res += 1
        return res